aten/src/ATen/native/BatchLinearAlgebra.cpp

```diff
@@ -3798,4 +3798,240 @@ Tensor _det_lu_based_helper_backward_helper(
+  //     # We need to clone A in this case. See [Cloning A]
+  //     clone A F-contig into A_f
+  //   else:
+  //     A_f = A;
+  // else:
+  //   clone A F-contig into A_f
+  // # Invariant: out_f is F-contig and A_f is F-ready
+  // # We pass FORTRAN the flags indicating if A_f is transposed and or conjugated
+  //
+  // # Here we undo the conjugations / transposes on out_f if needed
+  //
+  // if out_f not same out:
+  //   copy out_f into out
+  // return out
+  //
+  // Note: [Cloning A] If we are careful when allocating B when it needs to be allocated at the
+  // beginning of the algorithm, it is possible to always elide the copy of A here.
+  // Via this trick, the algorithm will copy at most one of A or B (never both) whenever A
+  // and B are F-ready (which happens almost always in practice).
+  // In most practical situations, this function copy at most one tensor.
+  // When called as f(A, B, out=B) in most practical cases it'll perform no copies.
+  //
+  // Note: The logic for the negative bit is the same as that for the conjugate bit
+
+  const bool avoid_copy_A = is_fortran_ready(A_broad) && A_broad.stride(-2) == 1 && A_broad.is_conj();
+  if (avoid_copy_A){
+    // See Note: [Cloning A]
+    at::native::resize_output(out, B_broad.sizes());
+  }
+  else {
+    // poorman's reimplementation of resize_output with result F-contig
+    if (resize_output_check(out, B_broad.sizes())) {
+      out.resize_(B_broad.transpose(-2, -1).sizes(), MemoryFormat::Contiguous);
+      out.transpose_(-2, -1);  // make 'out' have Fortran contiguous memory layout
+    }
+  }
+  // Invariant: out has the right size, so we'll be able to copy into it later on
+
+  Tensor out_f;
+  // We use C10_LIKELY mostly for documentation as it helps following what's the most likely path
+  if C10_LIKELY (is_fortran_ready(out)) {
+    out_f = out;
+    if C10_LIKELY (!out.is_same(B_broad)) {
+      out_f.copy_(B_broad);
+    }
+  } else{
+    if (avoid_copy_A){
+      // See Note: [Cloning A]
+      out_f = B_broad.clone(at::MemoryFormat::Contiguous);
+    }
+    else {
+      out_f = cloneBatchedColumnMajor(B_broad);
+    }
+  }
+  // Invariant: out_f F-ready and has B copied into it
+
+  // out_f is F-transposed
+  bool transpose_A = false;
+  bool transpose_out_f = false;
+  if (out_f.stride(-1) == 1) {
+    left = !left;
+    transpose_A = true;
+    transpose_out_f = true;
+    out_f.transpose_(-2 ,-1);
+  }
+
+  // No need to conjugate anything if out_f is conj as AX = conj(B) <=> conj(A)conj(X) = B
+  // and X = B after the algortihm. We just anotate that A is conjugated later on
+  // The solution will be written into out_f, so it'll be conjugated already
+
+  Tensor A_f;
+  bool A_is_conj = A_broad.is_conj() != out_f.is_conj();
+  bool A_is_neg = A_broad.is_neg() != out_f.is_neg();
+  bool A_is_f_contig = (A_broad.stride(-1) == 1) == transpose_A;
+  if C10_LIKELY (is_fortran_ready(A_broad) && !((A_is_conj && A_is_f_contig) || A_is_neg)) {
```

Comment:
`avoid_copy_A` should be used here?

Reply:
So, the idea of `avoid_copy_A`  is to be used whenever we need to allocate `B`. If we need to allocate `B`, we can choose to do so in column-major or row-major form, we get to choose. In this case, we look ahead, and make sure we allocate it in a way so that we do not ever enter the `else` branch here. In particular, it we see that choosing to allocate `B` F-contig we would get that `A_is_conj == true`  and `A_is_f_contig == true`, then we allocate `B` to be C-contig, which would make `A_is_f_contig == false` and would avoid to copy `A`.